### ISSUE TITLE
NaN results with inference RandLA-Net Tensorflow NPM3D

### ISSUE THREAD

**Issue by fredpeertje** [isl-org/Open3D-ML#323]
I get the following results when doing inference with the pretrained model on the Paris-lille-3D dataset. 

```
INFO - 2021-07-26 16:43:58,243 - parislille3d - Saved dijon_9 in ./test/ParisLille3D/dijon_9.txt.
test: 100%|██████████| 3/3 [06:02<00:00, 120.88s/it]
/usr/local/lib/python3.7/dist-packages/open3d/_ml3d/tf/modules/metrics/semseg_metric.py:53: RuntimeWarning: Mean of empty slice
  accs.append(np.nanmean(accs))
/usr/local/lib/python3.7/dist-packages/open3d/_ml3d/tf/modules/metrics/semseg_metric.py:86: RuntimeWarning: Mean of empty slice
  ious.append(np.nanmean(ious))
INFO - 2021-07-26 16:43:58,248 - semantic_segmentation - Per class Accuracy : [nan, nan, nan, nan, nan, nan, nan, nan, nan]
INFO - 2021-07-26 16:43:58,252 - semantic_segmentation - Per class IOUs : [nan, nan, nan, nan, nan, nan, nan, nan, nan]
INFO - 2021-07-26 16:43:58,254 - semantic_segmentation - Overall Accuracy : nan
INFO - 2021-07-26 16:43:58,257 - semantic_segmentation - Overall IOU : nan
```

**Comment by chrise96** [comment]
For the Toronto-3D issue:

There was a small bug in IoU calculation which is fixed in the `dev` branch. If you take mean of ious of each individual pointcloud in the test set, you get mIoU around 46. But what we want is mIoU obtained from the combined confusion matrix which comprises of whole test set. This will be fixed in the next release.
Toronto3D is a tricky dataset during training, some specific preprocessing is required to reproduce the results, I'll try to train and get back to you.

_Originally posted by @sanskar107 in https://github.com/isl-org/Open3D-ML/issues/256#issuecomment-819489869_

**Comment by jokokojote** [comment]
I see the issue is closed. How did you fix it? I get a similar result when running RandLA-Net with Tensorflow for SemanticKITTI.

Call:
`scripts/run_pipeline.py tf -c ml3d/configs/randlanet_semantickitti.yml --dataset.dataset_path Data/SemanticKITTI --pipeline SemanticSegmentation --dataset.use_cache True --split test --device gpu
`

Output (excerpt):
```
test: 100%|████████████████████████████| 20351/20351 [10:35:38<00:00,  1.87s/it]██████████████████████████████████████████████████████████████████████████████▌    | 84308/86731 [00:01<00:00, 55149.20it/s]
/home/pointclouduser/.pyenv/versions/3.8-dev/lib/python3.8/site-packages/open3d/_ml3d/tf/modules/metrics/semseg_metric.py:53: RuntimeWarning: Mean of empty slice
  accs.append(np.nanmean(accs))
/home/pointclouduser/.pyenv/versions/3.8-dev/lib/python3.8/site-packages/open3d/_ml3d/tf/modules/metrics/semseg_metric.py:86: RuntimeWarning: Mean of empty slice
  ious.append(np.nanmean(ious))
INFO - 2021-09-03 23:33:41,769 - semantic_segmentation - Per class Accuracy : [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan]
INFO - 2021-09-03 23:33:41,769 - semantic_segmentation - Per class IOUs : [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan]
INFO - 2021-09-03 23:33:41,769 - semantic_segmentation - Overall Accuracy : nan
INFO - 2021-09-03 23:33:41,769 - semantic_segmentation - Overall IOU : nan
```

**Comment by jokokojote** [comment]
I leave this here for everyone who is a little slow-witted like me: The SemanticKITTI test split does contain unlabeled data, so of course no performance can be calculated. I do not know about the ParisLille3D dataset @fredpeertje used.
